scripts/swupd_performance/swupd_logging_setup.bash

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/bash
+
+set -e
+
+DIRECTORY_PREFIX="/var/log/swupd"
+STAGING_DIRECTORY="$DIRECTORY_PREFIX"/staging/
+CONFIG_DIRECTORY="/etc/swupd_perform_update_config"
+
+check_for_root() {
+	if [ "$EUID" -ne 0 ]
+	then echo "Please run as root or with sudo"
+		exit
+	fi
+}
+
+
+
+setup_system() {
+	systemctl unmask --now swupd-update.service && echo "Unmasking swupd-update.service succeeded!"
+	systemctl unmask --now swupd-update.timer && echo "Unmasking swupd-update.timer succeeded!"
+	override_swupd_unit_file
+	setup_staging_directory
+	populate_config
+
+	echo ""
+	echo "Perform system daemon reload"
+	systemctl daemon-reload && echo "systemctl daemon-reload was successful!"
+	
+	echo ""
+	echo "Perform swupd-update.timer restart"
+	systemctl restart swupd-update.timer && echo "swupd-update.service was successfully scheduled!"
+}
+
+restore_system() {
+	user_input=
+	echo -n "Are you sure you want to clean files related to swupd performance (y/n)?"
+	read -r user_input
+	if [ "$user_input" != "y" ]; then
+		echo "Exiting program! No cleanup performed"
+		exit 0
+	fi
+
+	rm -r /etc/systemd/system/swupd-update.service || true
+	systemctl mask --now swupd-update.service && echo "Masking swupd-update.service succeeded!"
+	echo ""
+	systemctl mask --now swupd-update.timer && echo "Masking swupd-update.timer succeeded!"
+	echo ""
+
+	if [ -d "$CONFIG_DIRECTORY" ]; then
+		rm -rf "$CONFIG_DIRECTORY" && echo "Config directory removed"
+		echo ""
+	fi
+
+	if [ -d "$STAGING_DIRECTORY" ]; then
+		rm -rf "$STAGING_DIRECTORY" && echo "Staging directory & all the swupd update output files removed"
+		echo ""
+	fi
+
+	echo "System restored to original state"
+	exit 0
+}
+
+setup_staging_directory () {
+	echo "Setup staging directory"
+
+	if [ -d "$STAGING_DIRECTORY" ]; then
+		echo "Staging directory already exists"
+	else
+		mkdir -p "$STAGING_DIRECTORY" && echo "Staging directory: $STAGING_DIRECTORY successfully created!"
+	fi
+}
+
+populate_config() {
+	echo ""
+	echo "Setup performance monitoring script config dir"
+
+	if [ -d "$CONFIG_DIRECTORY" ]; then
+		echo "Config directory already exists"
+	else
+		mkdir $CONFIG_DIRECTORY && echo "Config directory created!"
```

Comment:
I'm not sure why this was not caught by shellcheck but you are missing double quotes around this variable in 3 places: line 80, 85 and 88

Reply:
I am guessing we don't need double quotes if we are using them inside multi-line strings